wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift

```diff
@@ -135,32 +135,32 @@ extension EventDecoder {
         _ events: [ZMUpdateEvent],
         startingAtIndex startIndex: Int64
     ) -> [ZMUpdateEvent] {
-        var decryptedEvents: [ZMUpdateEvent] = []
+        var decryptedEvents = [ZMUpdateEvent]()
+        let proteusService = syncMOC.proteusService!
 
-        // TODO: get core crypto lock
-        decryptedEvents = events.compactMap { event -> ZMUpdateEvent? in
-            switch event.type {
-            case .conversationOtrMessageAdd, .conversationOtrAssetAdd:
-                let proteusService = syncMOC.proteusService!
-
-                return decryptProteusEventAndAddClient(event, in: self.syncMOC) { sessionID, encryptedData in
-                    try proteusService.decrypt(
-                        data: encryptedData,
-                        forSession: sessionID
-                    )
-                }
+        proteusService.performBatchedOperations {
+            decryptedEvents = events.compactMap { event -> ZMUpdateEvent? in
+                switch event.type {
+                case .conversationOtrMessageAdd, .conversationOtrAssetAdd:
+                    return self.decryptProteusEventAndAddClient(event, in: self.syncMOC) { sessionID, encryptedData in
+                        try proteusService.decrypt(
+                            data: encryptedData,
+                            forSession: sessionID
+                        )
```

Comment:
isn't this attempting to acquire the lock too?

Reply:
Yes, but we discussed that we will test it first to see if it is a problem. It may not be given that it is being called in the same process and the same thread.